src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Contact.kt

```diff
@@ -0,0 +1,62 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.ContactType
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderBooking
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderContactPerson
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Person
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.ReferenceCode.Pk
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.RelationshipType
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderContactPersonRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.ReferenceCodeRepository
+
+@DslMarker
+annotation class OffenderContactPersonDslMarker
+
+@NomisDataDslMarker
+interface OffenderContactPersonDsl
+
+@Component
+class OffenderContactPersonBuilderRepository(
+  private val offenderContactPersonRepository: OffenderContactPersonRepository,
+  private val relationshipTypeRepository: ReferenceCodeRepository<RelationshipType>,
+  private val contactTypeRepository: ReferenceCodeRepository<ContactType>,
+) {
+  fun save(contact: OffenderContactPerson): OffenderContactPerson = offenderContactPersonRepository.save(contact)
+  fun lookupRelationshipType(code: String): RelationshipType = relationshipTypeRepository.findByIdOrNull(Pk(RelationshipType.RELATIONSHIP, code))!!
```

Comment:
would be better to call `getReferenceById` and then you wouldn't need the `!!`.  It would also throw a `EntityNotFoundException` rather than a null pointer too